tests/resources/test_resource_links.py

```diff
@@ -64,3 +60,18 @@ def test_update_links(app, client, input_data, headers):
 
     assert res.status_code == 200
     assert_expected_links(pid_value, res.json["links"])
+
+
+def test_config_hostname_links(app, client, input_data, headers):
+    site_hostname = "testsite.com"
+    app.config['SITE_HOSTNAME'] = site_hostname
```

Comment:
moderate: `app` is not a function level fixture, so we have to cleanup after ourselves:

```python
orig_site_hostname = app.config['SITE_HOSTNAME']
app.config['SITE_HOSTNAME'] = site_hostname

# rest of test ...

# cleanup
app.config['SITE_HOSTNAME'] = orig_site_hostname
```

Otherwise we get very strange behaviour.

Reply:
Nice catch!